src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,239 @@
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordsToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToInsert The sObjects to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> insertRecords(List<sObject> recordsToInsert, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordsToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert, Database.DMLOptions dmlOptions) {
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
```

Comment:
I think this should be AllowSave |= retrieveRespectDuplicateRuleSettings().  in other words, if the caller explicitly set AllowSave to true in dmlOptions (such as in Contact Merge), I don't think we want the global setting to turn it off.  But if the global setting is on, then it makes sense to force it on.

Reply:
That's a very good point.